screens/Ride/RideScreen.js

```diff
@@ -0,0 +1,294 @@
+  View,
+  Text,
+  TouchableOpacity,
+} from 'react-native';
+import { MapView } from 'expo';
+import Button from '../../components/Button';
+import { MaterialIcons } from '@expo/vector-icons';
+import { ThemeConsumer } from 'react-native-elements';
+// import { } from '../../components/Button';
+
+class RideScreen extends React.Component {
+  static navigationOptions = {
+    // title: 'Manual Mode',
+    header: null,
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      autopilotEnabled: false,
+      nextStopShift: false,
+      nextStopShiftReached: false,
+    };
+  }
+
+  render() {
+    const { navigation } = this.props;
+    const { autopilotEnabled, nextStopShift, nextStopShiftReached } = this.state;
+
+    return (
+      <View style={styles.container}>
+        <MapView
+          style={styles.mapPreview}
+          initialRegion={{
+            latitude: 52.5191406,
+            longitude: 13.4014149,
+            latitudeDelta: 0.1,
+            longitudeDelta: 0.1,
+          }}
+        />
+
+        <View
+          style={{
+            position: 'absolute',
+            top: 40,
+            left: '10%',
+            width: '80%',
+            backgroundColor: 'rgba(0, 0, 0, 0.8)',
+            borderRadius: 8,
+            padding: 12,
+            flexDirection: 'row',
+            alignItems: 'center',
+          }}
+        >
+          <MaterialIcons name="warning" size={36} style={{ color: '#ffffff', marginRight: 10, }} />
+          <View>
+            <Text style={{ fontSize: 19, color: '#ffffff' }}>Warning</Text>
+            <Text style={{ fontSize: 14, fontWeight: 'bold', color: '#ffffff' }}>A door is open</Text>
+          </View>
+        </View>
+
+        <View style={styles.content}>
+          {/* <Button
+            onPress={() => navigation.navigate('Control')}
+            title="Control"
+          />
+          <Button
+            onPress={() => navigation.navigate('RideCompletion')}
+            title="End Ride"
+          /> */}
+          <TouchableOpacity
+            onPress={() => {
+              this.setState({ nextStopShift: true, });
+              if (nextStopShift) {
+                this.setState({ nextStopShiftReached: true, });
+              }
+            }}
+          >
+            <View style={{
+              padding: 10,
+              backgroundColor: '#689FD9',
+            }}>
+              {!nextStopShift && (
+                <Text style={{ color: 'white' }}>Next: Pickup Passenger</Text>
+              )}
+              {nextStopShift && !nextStopShiftReached && (
+                <Text style={{ color: 'white' }}>Next: End Shift, drive to interchange point</Text>
+              )}
+              {nextStopShiftReached && (
+                <Text style={{ color: 'white' }}>Interchange point reached, end of shift</Text>
+              )}
+            </View>
+          </TouchableOpacity>
+
+          {!nextStopShiftReached && ((autopilotEnabled && (
```

Comment:
I had a bit of trouble with these conditions in JSX. Personally, I find these quite unreadable. In general, I would prefer to always refactor conditional JSX into custom render functions that then perform these checks.